src/merkle_tree.rs

```diff
@@ -0,0 +1,358 @@
+use sha2::{Digest, Sha256};
+
+pub struct MerkleTree {
+    tree: Vec<String>,
+    depth: u8,
```

Comment:
Have you considered getting `depth` from a method instead of storing it as an attribute and keeping track of it?